### PR TITLE
Add support for container-options

### PR DESCRIPTION
**Pull request by richm**
This adds support for `container-options` so that ansible-navigator
can pass through container options to ansible-runner.

### CODE REVIEW

**Comment by richm** [comment]
@nhosoi we won't be able to use ansible-navigator for our CI until this issue is resolved

**Comment by richm** [comment]
ping - can someone take a look at this?

File: ansible_navigator/actions/inventory.py [hunk 1]
```diff
@@ -440,7 +440,7 @@ def _collect_inventory_details(
         self,
     ) -> Tuple[Union[None, str], Union[None, str], Union[None, int]]:
 
-        # pylint:disable=too-many-branches
+        # pylint:disable=too-many-branches,too-many-statements
```
Comment (webknjaz) [inline ansible_navigator/actions/inventory.py:443]: The linters are there to prevent making the project unmaintainable. Please don't add ignores — they are supposed to be used as a last resort when it's impossible to refactor the code properly.

File: ansible_navigator/configuration_subsystem/navigator_post_processor.py [hunk 2]
```diff
@@ -297,6 +297,21 @@ def execution_environment_volume_mounts(
             entry.value.current = parsed_volume_mounts
         return messages, exit_messages
 
+    @staticmethod
+    @_post_processor
+    def container_options(
+        entry: Entry, config: ApplicationConfiguration
+    ) -> PostProcessorReturn:
+        # pylint: disable=unused-argument
```
Comment (webknjaz) [inline ansible_navigator/configuration_subsystem/navigator_post_processor.py:305]: This looks out-of-place. Normally, you should place ignores on lines violating them.

File: ansible_navigator/runner/api.py [hunk 3]
```diff
@@ -68,8 +68,8 @@ def __init__(
             container_volume_mounts ([list], optional): List of bind mounts in the form
                                                         ``host_dir:/container_dir:labels``.
                                                         Defaults to None.
-            container_options ([str], optional): List of container options to pass to execution
-                                                 engine. Defaults to None.
+            container_options ([list], optional): List of container options to pass to execution
```
Comment (webknjaz) [inline ansible_navigator/runner/api.py:71]: With the latest addition of Sphinx, this may fail on `main`. Plz rebase.

File: ansible_navigator/actions/inventory.py (hunk 1 above)
Comment (richm) [reply ansible_navigator/actions/inventory.py:443]: Too bad that wasn't enforced when the first person added too many branches and the `pylint:disable=too-many-branches` . . .

File: ansible_navigator/configuration_subsystem/navigator_post_processor.py (hunk 2 above)
Comment (richm) [reply ansible_navigator/configuration_subsystem/navigator_post_processor.py:305]: > This looks out-of-place. Normally, you should place ignores on lines violating them.

It conforms to the placement elsewhere in this file - all of the other methods that use the ignore have the ignore comment as the first line in the method - unless you want me to refactor the entire file to ensure that all of the pylint ignores used in the file are only placed on the lines where they are used

**Comment by richm** [comment]
note - the ansible/check failure is not a new failure - I get the same failure from the `main` branch code

**Comment by webknjaz** [comment]
I haven't seen that failure on main. It may be just flaky.

**Comment by webknjaz** [comment]
recheck

File: ansible_navigator/actions/inventory.py (hunk 1 above)
Comment (webknjaz) [reply ansible_navigator/actions/inventory.py]: And that's another problem. Sometimes it's okay to ignore things *temporarily* if you're going to be responsible for fixing that ASAP in a follow-up PR. But since not many people track such changes and forget about them, it's easier to just not introduce the linting violations in the initial patch.

File: ansible_navigator/configuration_subsystem/navigator_post_processor.py (hunk 2 above)
Comment (webknjaz) [reply ansible_navigator/configuration_subsystem/navigator_post_processor.py:303]: Ah, never mind. It's a wider problem then.

**Comment by ssbarnea** [comment]
zuul gate runs only when "gate" label is added and the result is that at the it merges the change.

File: test-requirements.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ansible-core
 black
+curses-2048
```
Comment (webknjaz) [inline test-requirements.txt:3]: Why did you add this?

File: ansible_navigator/actions/inventory.py [hunk 5]
```diff
@@ -435,12 +435,63 @@ def _build_inventory_list(self) -> None:
...
+        warn_msg = ["Errors were encountered while gathering the inventory:"]
+        warn_msg += inventory_err.splitlines()
+        self._logger.error(" ".join(warn_msg))
+        if "ERROR!" in inventory_err or "Error" in inventory_err:
+            warning = warning_notification(warn_msg)
+            self._interaction.ui.show(warning)
+        else:
+            self._extract_inventory(inventory_output)
+
+    def _collect_inventory_details_automated(
```
Comment (webknjaz) [inline ansible_navigator/actions/inventory.py:464]: Could you submit this refactoring as a separate PR so that it'd be easier to review this one with a smaller diff?

Comment (richm) [reply ansible_navigator/actions/inventory.py:464]: > Could you submit this refactoring as a separate PR so that it'd be easier to review this one with a smaller diff?

https://github.com/ansible/ansible-navigator/pull/563

File: test-requirements.txt (hunk 4 above)
Comment (richm) [reply test-requirements.txt]: I thought it would fix one of the pytest failures, but apparently not

File: docs/settings.rst [hunk 6]
```diff
@@ -215,6 +217,19 @@ The following table describes all available settings.
               execution-environment:
                 container-engine:
 
+  * - container-options
```
Comment (webknjaz) [inline docs/settings.rst:220]: This hardcoded table is being removed in #564 which will probably be auto-merged within an hour. So there's no need to update this file anymore. You'll need to exclude it from the next rebase.

I don't have any further comments on this PR but I'd like for my other teammates to also look at it first before merging.
